### PR TITLE
[native] Refactor server operation endpoint

### PR DESCRIPTION
**Pull request by tanjialiang**
* Let task/listAll endpoint to return valid json format
* Let task/listAll endpoint print meaningful error message when 'limit' is malformatted
* Let systemConfig/getProperty endpoint print meaningful error message if property does not exist.
```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Review by xiaoxmeng — APPROVED** [review]
@tanjialiang LGTM. Thanks for the quick fix!

File: presto-native-execution/presto_cpp/main/PrestoServerOperations.cpp [hunk 1]
```diff
@@ -207,28 +207,27 @@ std::string PrestoServerOperations::taskOperation(
...
       oss << "[";
       uint32_t count = 0;
-      for (const auto& task : taskMap) {
-        const auto& veloxTask = task.second->task;
+      for (auto taskItr = taskMap.begin(); taskItr != taskMap.end();
+           ++taskItr) {
+        const auto& veloxTask = taskItr->second->task;
         if (++count > limit) {
-          oss << "... " << (taskMap.size() - limit) << " more tasks ...\n";
+          oss << "... " << (taskMap.size() - limit) << " more tasks ...]\n";
```
Comment (mbasmanova) [inline presto-native-execution/presto_cpp/main/PrestoServerOperations.cpp:225]: Curious, how do you verify that this returns valid JSON. Would you share an example response? It feels like it is not a valid JSON.

File: presto-native-execution/presto_cpp/main/PrestoServerOperations.cpp [hunk 2]
```diff
@@ -143,7 +143,7 @@ std::string PrestoServerOperations::systemConfigOperation(
       return fmt::format(
           "{}\n",
           SystemConfig::instance()->optionalProperty(name).value_or(
-              "<default>"));
+              fmt::format("Could not find property '{}'", name)));
```
Comment (mbasmanova) [inline presto-native-execution/presto_cpp/main/PrestoServerOperations.cpp:146]: Can we make sure to return HTTP 500 on errors?

**Review by mbasmanova — COMMENTED** [review]
Some questions.

File: presto-native-execution/presto_cpp/main/PrestoServerOperations.cpp (hunk 1 above)
Comment (tanjialiang) [reply presto-native-execution/presto_cpp/main/PrestoServerOperations.cpp:225]: When limit is put, the option is to return an incomplete JSON with the ... n more tasks ... at the end to indicate the whole picture.

File: presto-native-execution/presto_cpp/main/PrestoServerOperations.cpp (hunk 2 above)
Comment (tanjialiang) [reply presto-native-execution/presto_cpp/main/PrestoServerOperations.cpp:146]: Yeah we can do that

File: presto-native-execution/presto_cpp/main/PrestoServerOperations.cpp (hunk 1 above)
Comment (tanjialiang) [reply presto-native-execution/presto_cpp/main/PrestoServerOperations.cpp:225]: But we can probably change it to return a JSON with limited tasks in it and a trailing statement saying n more tasks

**Review by xiaoxmeng — APPROVED** [review]

**Pull request merged**